src/client.js

```diff
@@ -22,6 +31,22 @@ class Client {
         if (typeof payload !== "undefined") Object.assign(data, payload)
         this.socket.write(JSON.stringify(data) + "\n")
     }
+
+    // for proper socket closing
+    close() {
+        this.socket.end()
+    }
+
+    // getconnection establishes connection based on underlying OS
+    getconnection() {
```

Comment:
Could you rename this function to `connect`?

Reply:
renamed to connect()